File: Week 1/mario-more.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Запрашиваем высоту пирамиды, при условии [1;8]
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);
    // Цикл количества строк и столбцов
    for (int i = 0; i < n; i++)
    {
        // Цикл пустых столбиков для пирамиды №1
        for (int j = 0; j < n - i - 1; j++)
        {
            printf(" ");
        }
        // Цикл столбиков # для пирамиды №1
        for (int z = 0; z < i + 1; z++)
        {
            printf("#");
        }
        printf("  ");
        // Цикл столбиков # для пирамиды №2
        for (int k = 0; k <= i; k++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: Week 6/mario more.py
from cs50 import get_int


def main():
    height = get_height()
    for col in range(1, height + 1):
        print_pyramid1(height, col)
        print_pyramid2(height, col)
# Print pyramids


def print_pyramid1(n, col):
    print(" " * (n - col) + "#" * col, end="  ")


def print_pyramid2(n, col):
    print("#" * col, end="\n")
# Input validation


def get_height():
    while True:
        h = get_int("Height: ")
        if 1 <= h <= 8:
            break
    return h


main()